realm/realm-library/src/main/java/io/realm/RealmChangeListener.java

```diff
@@ -29,19 +29,19 @@
  * for that background thread.
  * <p>
  * All {@link io.realm.RealmObject} and {@link io.realm.RealmResults} will automatically contain their new values when
- * the {@link #onChange()} method is called. Normally this means that it isn't necessary to query again for those
+ * the {@link #onChange(Object)} method is called. Normally this means that it isn't necessary to query again for those
  * objects, but just invalidate any UI elements that are using them. If there is a chance that a object has been been
  * deleted, it can be verified by using {@link RealmObject#isValid()}.
  *
  * @see Realm#addChangeListener(RealmChangeListener)
  * @see Realm#removeAllChangeListeners()
  * @see Realm#removeChangeListener(RealmChangeListener)
  */
-public interface RealmChangeListener {
+public interface RealmChangeListener<T> {
 
     /**
      * Called when a transaction is committed.
      */
```

Comment:
Missing `@param` description, I think we especially need to describe that it is live objects being returned and you shouldn't store them and except to be able to compare them against the "next" value.